pysnooper/pysnooper.py

```
@@ -20,26 +21,31 @@
+                s = s.decode('utf-8')
             stderr = sys.stderr
             stderr.write(s)
-            stderr.write('\n')
     elif isinstance(output, (pycompat.PathLike, str)):
-        output_path = pathlib.Path(output)
         def write(s):
-            with output_path.open('a') as output_file:
+            s += '\n'
+            if isinstance(s, bytes): # Python 2 compatibility
+                s = s.decode('utf-8')
+            with open(output_path, 'a') as output_file:
```

Ruff (0.12.2)

Undefined name `output_path`

(F821)